crates/pop-cli/src/commands/call/chain.rs

```diff
@@ -222,11 +250,36 @@ impl CallChainCommand {
 		call_data: &str,
 		cli: &mut impl Cli,
 	) -> Result<()> {
-		// Resolve who is signing the extrinsic.
+		// Resolve who is signing the extrinsic. If a `suri` was provided via the command line,
+		// skip the prompt.
+		let mut use_wallet = self.use_wallet;
 		let suri = match self.suri.as_ref() {
-			Some(suri) => suri,
-			None => &cli.input("Signer of the extrinsic:").default_input(DEFAULT_URI).interact()?,
+			Some(suri) => suri.clone(),
+			None =>
+				if !self.use_wallet {
```

Comment:
Seems to be very similar code as above - can we refactor into a function?

Reply:
refactored into `determine_signing_method`: https://github.com/r0gue-io/pop-cli/pull/371/commits/6035fd86b7ca679127b5cd34765a51dca6a66f13